### ISSUE TITLE
methylSigReadData bug

### ISSUE THREAD

**Issue by JLGlass** [sartorlab/methylSig#22]
When running 

meth <- methylSigReadData(vec.files, pData=df.pData, assembly='hg19', destranded=T, num.cores=8)

I get the following error:

Error in bsseq::read.bismark(files = fileList, sampleNames = rownames(pData),  : 
  unused arguments (sampleNames = rownames(pData), fileType = fileType, mc.cores = num.cores)

Running this same code ~1 month ago didn't cause this issue, so I suspect it's due to an update in bsseq::read.bismark?

**Comment by rcavalcante** [comment]
Hi,

Your suspicion is very likely correct. There was some point at which `bsseq::read.bismark` dropped the `sampleNames` and `pData` arguments in favor of `colData`. They also dropped teh `fileType`.

I wonder, when your code was working, were you using R 3.4.z and Bioconductor 3.6? And now that your code is broken, have you upgraded your version of R to 3.5.z and are using Bioconductor 3.8. Or some similar situation?

The current release of `methylSig` is only meant to work in R 3.4.z with Bioconductor 3.6 for this reason, in addition to an update to `bsseq` that doesn't allow the `GRanges` object in their `BSseq` class to have widths > 1, which destroys the tiling functionality of `methylSig`.

Long story short, if it's possible for you to go back to the R 3.4.z and Bioconductor 3.6 ecosystem, I would recommend that for now (Docker or Singularity is a nice option and I can help you with a `Dockerfile` if you're unfamiliar with setting that up). 

Otherwise, getting `methylSig` to play nice with the updated `bsseq` versions is on my to do list (sometime in the next two weeks actually), so if time isn't of the essence, I can hopefully get it fixed for you.

Thanks for using `methylSig`, and for reaching out with your problem.

Raymond

**Comment by JLGlass** [comment]
Hi Raymond,

Thank you for the quick reply. That sounds exactly right - I updated bioconductor relatively recently. If you have a dockerfile, that would be very helpful. I'd be happy to test out the code when you've updated it.

Best,

Jacob

**Comment by rcavalcante** [comment]
I think this should do it:

https://github.com/rcavalcante/dockers/tree/master/bioc_release_36

You may want to edit the Bioconductor packages to make it install faster. This is the image I keep around to deal with problems in the packages I support so there's a bunch of extraneous stuff to `methylSig`, e.g. `annotatr`, `chipenrich`, etc.

**Comment by jsl26** [comment]
Hi Raymond, 

Have run into the same problem as Jacob and was wondering if you had a chance to look at the issue? 

thanks

joe

**Comment by bhanratt** [comment]
It is now August 2019 and I am getting the same issue as the above users. Should we consider this package no longer supported going forward or will it be updated?

**Comment by rcavalcante** [comment]
Hello,

This is open source software and I welcome pull requests. A roadblock to updating the package was that the bsseq package dependency changed a behavior that we really relied on.

With that said, there is a docker image that I've been using when I need to use methylSig (https://hub.docker.com/r/umichbfxcore/r_344).

I do want to get methylSig working with the most current version of R, and I hope to dedicate time to that before the summer is out. Other priorities got in the way of this project.

Thanks.

**Comment by sixone11** [comment]
Summer is out.... i feel so sad that it still can't be used in newer R version.

**Comment by rcavalcante** [comment]
Hi, this is what I'm going to work on this week.

**Comment by marin-e** [comment]
Hello, 

I have the same issue as the above users. Will methylSig be updated soon?

Thank you.

**Comment by marin-e** [comment]
Hello, I successfully ran methylSig using your docker image but I failed to run methylSigTile and methylSigAnnotation because the hg38 assembly version is not supported with these versions of annotatr and GenomeInfoDb. Do you have an idea of how solving this?

Thank you

**Comment by rcavalcante** [comment]
Hello all,

In anticipation of submitting the package to Bioconductor, I have done some refactoring of the methylSig code base.

Please see the vignette which is included as part of the README.md for details. The package nwo works with R 3.6 and Bioc 3.10 as well as in the devel versions of Bioc.

Please give it a try. For now, I will close this issue. Please open a new issue with problems relating to the new release (https://github.com/sartorlab/methylSig/releases/tag/v0.99.0).
